src/components/DetailsMetrics/DetailsMetrics.js

```diff
@@ -248,7 +255,13 @@ const DetailsMetrics = ({ selectedItem }) => {
   )
 
   useEffect(() => {
+    if (selectedItem.name !== prevSelectedEndPointNameRef.current) {
+      prevSelectedEndPointNameRef.current = selectedItem.name
+      setMetrics([])
+      return
+    }
```

Comment:
Why do you clear metrics here if usually it's being done in the cleanup function that is return from the useEffect? You don't need to set the model endpoint name to variable and then check it. You just need to reset metrics on changing `selectedItem`

Reply:
@mavdryk change the validation to:
   if (selectedItem.metadata.uid !== prevSelectedEndPointNameRef.current) {
      prevSelectedEndPointNameRef.current = selectedItem.metadata.uid
      return
    }
    
The prevSelectedEndPointNameRef shall prevent redundant invocation requests while the user selects the endpoint from the list.